### PR TITLE
test: use junit assert instead of java assert statement in test

### PR DESCRIPTION
**Pull request by shoothzj**
### Motivation

See #995

### Changes

1. use junit assert instead of java assert statement in test
2. delete a unused util class also uses assert statement

### CODE REVIEW

**Comment by shoothzj** [comment]
@dlg99 @eolivelli @hangc0276 @merlimat @nicoloboschi @zhaijack PTAL

**Comment by shoothzj** [comment]
@dlg99 @eolivelli @hangc0276 @merlimat @nicoloboschi @zhaijack PTAL

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/util/DaemonThreadFactory.java
Comment (jiazhai) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/util/DaemonThreadFactory.java:1]: Seems this change is not related with the target. Is this class not used anywhere?  why this is removed?

Comment (shoothzj) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/util/DaemonThreadFactory.java:1]: @jiazhai This class is not used anywhere even in pulsar project. And this class also uses assert statement.

**Review by jiazhai — APPROVED** [review]

**Comment by hangc0276** [comment]
This PR can only be released in major releases because we removed DaemonThreadFactory in the util package and it may be used in other projects.
